python/semantic_kernel/connectors/search/brave/brave_search.py

```diff
@@ -0,0 +1,210 @@
+# Copyright (c) Microsoft. All rights reserved.
+
+import logging
+from collections.abc import AsyncIterable
+from typing import TYPE_CHECKING, Any
+
+from httpx import AsyncClient, HTTPStatusError, RequestError
+from pydantic import ValidationError
+
+from semantic_kernel.connectors.search.brave.brave_search_response import BraveSearchResponse
+from semantic_kernel.connectors.search.brave.brave_search_settings import BraveSettings
+from semantic_kernel.connectors.search.brave.brave_web_page import BraveWebPage
+from semantic_kernel.connectors.search.brave.const import DEFAULT_URL, QUERY_PARAMETERS
+from semantic_kernel.data.text_search import (
+    AnyTagsEqualTo,
+    EqualTo,
+    KernelSearchResults,
+    SearchFilter,
+    TextSearch,
+    TextSearchOptions,
+    TextSearchResult,
+)
+from semantic_kernel.exceptions import ServiceInitializationError, ServiceInvalidRequestError
+from semantic_kernel.kernel_pydantic import KernelBaseModel
+from semantic_kernel.utils.feature_stage_decorator import experimental
+from semantic_kernel.utils.telemetry.user_agent import SEMANTIC_KERNEL_USER_AGENT
+
+if TYPE_CHECKING:
+    from semantic_kernel.data.text_search import SearchOptions
+
+logger: logging.Logger = logging.getLogger(__name__)
+
+
+@experimental
+class BraveSearch(KernelBaseModel, TextSearch):
+    """A search engine connector that uses the Brave Search API to perform a web search."""
+
+    settings: BraveSettings
+
+    def __init__(
+        self,
+        api_key: str | None = None,
+        env_file_path: str | None = None,
+        env_file_encoding: str | None = None,
+    ) -> None:
+        """Initializes a new instance of the Brave Search class.
+
+        Args:
+            api_key: The Brave Search API key. If provided, will override
+                the value in the env vars or .env file.
+            env_file_path: The optional path to the .env file. If provided,
+                the settings are read from this file path location.
+            env_file_encoding: The optional encoding of the .env file. If provided,
+                the settings are read from this file path location.
+        """
+        try:
+            settings = BraveSettings.create(
```

Comment:
.create is no longer necessary:
```suggestion
            settings = BraveSettings(
```

Reply:
@eavanvalkenburg  Updated the method as requested
```
            settings = BraveSettings(
                api_key=api_key,
                env_file_path=env_file_path,
                env_file_encoding=env_file_encoding,
            )
```